### PR TITLE
NMS-14174: working on example scripts

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-14174

Scripts added for:
* Cisco IOS
* Juniper
* ArubaOS & ArubaOS-CX
* Palo Alto

### CODE REVIEW

**Review by fooker — DISMISSED** [review]

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 1]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+.Example scripts for Aruba devices
+[options="header" cols="1,2,3"]
+|===
+| Script
+| Target device
+| Description
+
+| aruba-cx-cli.dcb
+| Aruba OS-CX
+| retrieves config from ArubaOS-CX device in cli format
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:86]:
```suggestion
| Retrieves config from ArubaOS-CX device in CLI format.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 2]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| aruba-cx-json.dcb
+| Aruba OS-CX
+| retrieves config from ArubaOS-CX device in json format
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:90]:
```suggestion
| Retrieves config from ArubaOS-CX device in JSON format.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 3]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| aruba-os.dcb
+| Aruba OS
+| retrieves config from ArubaOS device
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:94]:
```suggestion
| Retrieves config from ArubaOS device.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 4]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| cisco-ios-running.dcb
+| Cisco IOS
+| retrieves running-config from Cisco IOS device
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:98]:
```suggestion
| Retrieves running-config from Cisco IOS device.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 5]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| cisco-ios-startup.dcb
+| Cisco IOS
+| retrieves startup-config from Cisco IOS device
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:102]:
```suggestion
| Retrieves startup-config from Cisco IOS device.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 6]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| juniper-junos-config-gz.dcb
+| Juniper JunOS
+| retrieves compressed configuration from Juniper JunOS device
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:106]:
```suggestion
| Retrieves compressed configuration from Juniper Junos device.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 7]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| juniper-junos-config-set.dcb
+| Juniper JunOS
+| retrieves configuration from Juniper JunOS device in set-format
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:110]:
```suggestion
| Retrieves configuration from Juniper Junos device in set-format.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 8]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| juniper-junos-config-txt.dcb
+| Juniper JunOS
+| retrieves uncompressed configuration from Juniper JunOS device
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:114]:
```suggestion
| Retrieves uncompressed configuration from Juniper Junos device.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 9]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+
+| paloalto-panos-config.dcb
+| Palo Alto PanOS
+| retrieves configuration from Palo Alto PanOS device
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:118]:
```suggestion
| Retrieves configuration from Palo Alto PAN-OS device.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 10]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+|===
+
+=== Configuring DCB user accounts
+
+We strongly recommend limiting the privileges of the user accounts used for configuration retrieval.
+The following examples demonstrate, how to configure a `dcb` user account, specifying its password `Pa55w0rd` and limiting its privileges to configuration retrieval commands.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:124]:
```suggestion
The following examples demonstrate how to configure a `dcb` user account, specifying its password `Pa55w0rd` and limiting its privileges to configuration retrieval commands.
```

File: docs/modules/operation/pages/device-config-backup/ssh.adoc [hunk 11]
```diff
@@ -46,26 +46,154 @@ In addition, you must append the script variable `filenameSuffix` to the filenam
...
+[source, script]
+----
+cisco-ios#configure terminal
+cisco-ios(config)#username dcb privilege 2 secret Pa55w0rd
+cisco-ios(config)#privilege exec level 2 copy running-config
+cisco-ios(config)#privilege exec level 2 copy startup-config
+cisco-ios(config)#exit
+----
+
+==== Juniper JunOS
```
Comment (Bonrob2) [inline docs/modules/operation/pages/device-config-backup/ssh.adoc:173]:
```suggestion
==== Juniper Junos
```

**Review by Bonrob2 — DISMISSED** [review]
Minor changes to the docs. Thanks for adding the examples and documenting them.

**Review by fooker — APPROVED** [review]

**Pull request merged**